client/my-sites/email/form/mailboxes/components/field.tsx

```diff
@@ -0,0 +1,116 @@
+import { FormInputValidation } from '@automattic/components';
+import { useState } from 'react';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormPasswordInput from 'calypso/components/forms/form-password-input';
+import FormTextInput from 'calypso/components/forms/form-text-input';
+import FormTextInputWithAffixes from 'calypso/components/forms/form-text-input-with-affixes';
+import { useGetDefaultFieldLabelText } from 'calypso/my-sites/email/form/mailboxes/components/use-get-default-field-label-text';
+import type {
+	MailboxFormFieldBase,
+	MutableFormFieldNames,
+} from 'calypso/my-sites/email/form/mailboxes/types';
+import type { ChangeEvent } from 'react';
+
+interface MailboxFormFieldProps {
+	domains?: string[];
```

Comment:
The most common use case will be to use a single domain, so probably we should add a compound type:
```suggestion
	domains?: string[] | string;
```

Reply:
I think I forgot to remove this. It's never used anywhere at this level/stage.

> The most common use case will be to use a single domain

Actually the common use case is to pass the `selectedDomainName`.  In this case the `field` object already contains the active domain name, so other props denoting that are redundant.

We often additionally add the `domains[]` prop when we need to reckon with multiple domains, like when the user can switch between the active domain i.e. `selectedDomainName`